RELEASE_NOTE_DRAFT.md

```diff
@@ -1,37 +1,106 @@
 10.0.0
 ====================
 
-This is the 10.0.0 major release.
+This is the 10.0.0 major release. 
+Central theme of this release is addition of new APIs, features as requested by the community and decrease Tech Debt. 
+This is our yearly major release.
 
-New Functionality
+# New Functionality
 -----------------
-
-* 
-
-Backwards Breaking Changes
+* Changed collection factories to be services that get loaded by the ServiceLoader in the API package.
+* Changed RichIterable.groupByUniqueKey() so that it's target extends MutableMapIterable instead of MutableMap.
+* Implemented UnmodifiableMutableOrderedMap.
+* Added <primitive1><primitive2>To<primitive1>Function.
+* Added specialized MultiReader interfaces for List, Set and Bag to improve the interface hierarchy.
+* Implemented RichIterable.getAny().
+* Implemented RichIterable.countByEach().
```

Comment:
countByEach is implemented as default methods on RichIterable so should not be a breaking change